app/src/main/java/org/metabrainz/mobile/activity/ArtistActivity.java

```diff
@@ -71,15 +70,41 @@ private void findViews() {
         recyclerView = findViewById(R.id.recycler_view);
     }
 
-    private void setArtist(Artist data){
-        if(data != null){
-            artistViewModel.setArtist(data);
-            Log.d(LOG_TAG,data.getName());
-            setArtistInfo();
+    private void setArtist(Artist artist){
+        if (artist != null){
+            getSupportActionBar().setTitle(artist.getName());
+
+            getArtistWiki(artist);
+
+            String type = artist.getType();
+            if (type != null && !type.isEmpty()) {
+                artistType.setText(type);
+            }
+
+            String gender = artist.getGender();
+            if (gender != null && !gender.isEmpty()) {
+                artistGender.setText(gender);
+            }
+
+            if (artist.getArea() != null) {
+                artistArea.setText(artist.getArea().getName());
+            }
+
+            if (artist.getLifeSpan() != null) {
+                artistLifeSpan.setText(artist.getLifeSpan().getTimePeriod());
+            }
+
+            if (artist.getReleases() != null){
+                // Maybe releases get updated with cover art info
+                // Notify the adapter that the info is new
+                releaseList.clear();
+                releaseList.addAll(artist.getReleases());
+                adapter.notifyDataSetChanged();
```

Comment:
As the entire list gets updated with each call, I think we should move on to using `DiffUtil` to avoid overheads.